gpflow/likelihoods/base.py

```diff
@@ -48,97 +50,119 @@ def __init__(self, latent_dim: Optional[int], observation_dim: Optional[int]) ->
         :param observation_dim: the dimension of the observation vector Y for a single data point
         """
         super().__init__()
+        self.input_dim = input_dim
         self.latent_dim = latent_dim
         self.observation_dim = observation_dim
 
-    def _check_last_dims_valid(self, F: TensorType, Y: TensorType) -> None:
+    def _check_last_dims_valid(self, X: TensorType, F: TensorType, Y: TensorType) -> None:
```

Comment:
Can you explain why you choose this order for the input args?

Reply:
> Can you explain why you choose this order for the input args?

It seems most intuitive to me. You start with an X, then you map that to F, then you map that to Y?